### ISSUE TITLE
Strange behaviour `dd.from_pandas(df)` -  changes dtype (object -> string)

### ISSUE THREAD

**Issue by vanheck** [dask/dask#11720]
I grab the solution for [one-hot encoding with multilabels](https://stackoverflow.com/a/70000569/6089311). And I tried to change it for numerical labels:

```py
import dask.dataframe as dd
import pandas as pd
from sklearn.preprocessing import MultiLabelBinarizer

df = pd.DataFrame({"col_a": [[1, 2, 3], [2], [1], [4, 2], [2, 1]]})
all_classes = df["col_a"].explode().unique()
mlb = MultiLabelBinarizer(classes=all_classes)

def onehot_multilabel(df):
    return pd.DataFrame(mlb.fit_transform(df["col_a"].values), columns=all_classes)

ddf = dd.from_pandas(df, npartitions=2)
ddf.map_partitions(onehot_multilabel, meta=pd.DataFrame(columns=all_classes)).compute()
```

output:
```
   1  2  3  4
0  0  0  0  0
1  0  0  0  0
2  0  0  0  0
0  0  0  0  0
1  0  0  0  0
```

The output here is wrong (all are zeros), because in pandas `df["col_a"]` has dtype object. But `dd.from_pandas(df)` changes the dtype of the column to `string`, why? I have any idea.

**Anything else we need to know?**:

If I dont use `dd.from_pandas()`, but seriealize to parquet file, it works well:
```py
from tempfile import NamedTemporaryFile
import dask.dataframe as dd
import pandas as pd
from sklearn.preprocessing import MultiLabelBinarizer

df = pd.DataFrame({"col_a": [[1, 2, 3], [2], [1], [4, 2], [2, 1]]})
all_classes = df["col_a"].explode().unique()
mlb = MultiLabelBinarizer(classes=all_classes)
def onehot_multilabel(df):
    return pd.DataFrame(mlb.fit_transform(df["col_a"].values), columns=all_classes)

with NamedTemporaryFile(suffix=".parquet") as parquetfile:
    df.to_parquet(parquetfile.name)
    ddf = dd.read_parquet(parquetfile.name)
    result = ddf.map_partitions(
        onehot_multilabel, meta=pd.DataFrame(columns=all_classes)
    ).compute()
result
```

output:
```
   1  2  3  4
0  1  1  1  0
1  0  1  0  0
2  1  0  0  0
3  0  1  0  1
4  1  1  0  0
```

**Environment**:

- Pandas: 2.1.4
- Dask version: 2025.1.0
- dask-expr: 1.1.21
- scikit-learn: 1.6.0
- Python version: 3.11.11 (main, Dec 11 2024, 16:28:39) [GCC 11.2.0]
- Operating System: Linux 6.1.0-28-amd64 # 1 SMP PREEMPT_DYNAMIC Debian 6.1.119-1 x86_64 x86_64 x86_64 GNU/Linux
- Install method (conda, pip, source): conda-env (installation with pip)

**Comment by phofl** [comment]
Thanks for your report. This is expected unfortunately in thee cases, see https://github.com/dask/dask/issues/10139 for more information. You can disable this option if using from_pandas is necessary for your workload
